gcalbot/gcalbot/handler.go

```diff
@@ -0,0 +1,85 @@
+package gcalbot
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat"
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/chat1"
+	"github.com/keybase/managed-bots/base"
+	"golang.org/x/oauth2"
+)
+
+type Handler struct {
+	*base.DebugOutput
+
+	kbc      *kbchat.API
+	db       *DB
+	requests *base.OAuthRequests
+	config   *oauth2.Config
+}
+
+var _ base.Handler = (*Handler)(nil)
+
+func NewHandler(kbc *kbchat.API, db *DB, requests *base.OAuthRequests, config *oauth2.Config) *Handler {
+	return &Handler{
+		DebugOutput: base.NewDebugOutput("Handler", kbc),
+		kbc:         kbc,
+		db:          db,
+		requests:    requests,
+		config:      config,
+	}
+}
+
+func (h *Handler) HandleNewConv(conv chat1.ConvSummary) error {
+	welcomeMsg := "Hello! I can get you setup with a Google Meet video call anytime, just send me `!meet`."
+	return base.HandleNewTeam(h.DebugOutput, h.kbc, conv, welcomeMsg)
+}
+
+func (h *Handler) HandleCommand(msg chat1.MsgSummary) error {
+	if msg.Content.Text == nil {
+		h.Debug("skipping non-text message")
+		return nil
+	}
+
+	cmd := strings.TrimSpace(msg.Content.Text.Body)
+
+	if !strings.HasPrefix(cmd, "!gcal") {
+		return nil
+	}
+
+	tokens, err := base.TokensFromCommand(cmd)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case strings.HasPrefix(cmd, "!gcal accounts"):
+		if !(msg.Sender.Username == msg.Channel.Name) {
+			_, err = h.kbc.SendMessageByConvID(msg.ConvID, "All `!gcal accounts` commands must be sent over direct message.")
```

Comment:
why restrict to DM here?

Reply:
I thought maybe it would be a good idea because `!gcal accounts` can expose some user data, but now I'm a lot more unsure because all of the commands could expose user data anyways 😛

Reply:
I think it's ok to reference in a team setting